engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -908,7 +908,7 @@ private void execute() throws YarnException, IOException
           LOG.info("Releasing {} as resource with priority {} was already assigned", allocatedContainer.getId(), allocatedContainer.getPriority());
           releasedContainers.add(allocatedContainer.getId());
           numReleasedContainers++;
-          numRequestedContainers--;
+          numRequestedContainers++;    // undo the decrement above for this allocated container
```

Comment:
@sanjaypujare
- The change needs to be tested otherwise it can't be merged.
- While refactoring can be done as part of a separate JIRA, for this change to be merged, it is necessary to review and understand the logic for counting requested and released containers. I am -1 for any changes where a contributor does not know for sure.